plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -2047,6 +2047,43 @@ fc::variant read_only::get_block(const read_only::get_block_params& params) cons
            ("ref_block_prefix", ref_block_prefix);
 }
 
+fc::variant read_only::get_block_header(const read_only::get_block_header_params& params) const {
+   signed_block_ptr block;
+   optional<uint64_t> block_num;
+
+   EOS_ASSERT( !params.block_num_or_id.empty() && params.block_num_or_id.size() <= 64,
+               chain::block_id_type_exception,
+               "Invalid Block number or ID, must be greater than 0 and less than 64 characters"
+   );
+
+   try {
+      block_num = fc::to_uint64(params.block_num_or_id);
+   } catch( ... ) {}
+
+   if( block_num.valid() ) {
+      block = db.fetch_block_by_number( *block_num );
+   } else {
+      try {
+         block = db.fetch_block_by_id( fc::variant(params.block_num_or_id).as<block_id_type>() );
+      } EOS_RETHROW_EXCEPTIONS(chain::block_id_type_exception, "Invalid block ID: ${block_num_or_id}", ("block_num_or_id", params.block_num_or_id))
+   }
+
+   EOS_ASSERT( block, unknown_block_exception, "Could not find block: ${block}", ("block", params.block_num_or_id));
+
+   return fc::mutable_variant_object ()
+         ("block_num", block->block_num())
+         ("id", block->id())
+         ("timestamp", block->timestamp)
+         ("producer", block->producer)
+         ("confirmed", block->confirmed)
+         ("previous", block->previous)
+         ("transaction_mroot", block->transaction_mroot)
+         ("action_mroot", block->action_mroot)
+         ("schedule_version", block->schedule_version)
+         ("producer_signature", block->producer_signature)
```

Comment:
Is there a reason for writing these out instead of upcasting the block to signed_block_header?  You're not including `header_extensions` or `new_producers`

Reply:
according to the slack chat with todd, it is the expected output. Will double-check to clarify the requirements.

Reply:
Don't want `header_extensions` since it could be large in the future. Don't need `new_producers` and it simplifies the history-tools implementation to not have that.

Reply:
I strongly disagree with this reasoning.  An api called get_block_header should return the whole block header.

Reply:
Also, I don't buy the argument that the header extensions "could be large in the future."  We shouldn't put anything into the block header extensions unless it really needs to be part of the header.